lib/language_pack/helpers/bundler_wrapper.rb

```diff
@@ -1,37 +1,89 @@
+# frozen_string_literal: true
+
 require 'language_pack/fetcher'
 
+# This class is responsible for installing and maintaining a
+# reference to bundler. It contains access to bundler internals
+# that are used to introspect a project such as detecting presence
+# of gems and their versions.
+#
+# Example:
+#
+#   bundler = LanguagePack::Helpers::BundlerWrapper.new
+#   bundler.install
+#   bundler.version                 => "1.15.2"
+#   bundler.dir_name                => "bundler-1.15.2"
+#   bundler.has_gem?("railties")    => true
+#   bundler.gem_version("railties") => "5.2.2"
+#   bundler.clean
+#
+# Also used to determine the version of Ruby that a project is using
+# based on `bundle platform --ruby`
+#
+#   bundler.ruby_version # => "ruby-2.5.1"
+#   bundler.clean
+#
+# IMPORTANT: Calling `BundlerWrapper#install` on this class mutates the environment variable
+# ENV['BUNDLE_GEMFILE']. If you're calling in a test context (or anything outside)
+# of an isolated dyno, you must call `BundlerWrapper#clean`. To reset the environment
+# variable:
+#
+#   bundler = LanguagePack::Helpers::BundlerWrapper.new
+#   bundler.install
+#   bundler.clean # <========== IMPORTANT =============
+#
 class LanguagePack::Helpers::BundlerWrapper
   include LanguagePack::ShellHelpers
 
+  BLESSED_BUNDLER_VERSIONS = {}
+  BLESSED_BUNDLER_VERSIONS["1"] = "1.15.2"
+  BLESSED_BUNDLER_VERSIONS["2"] = "2.0.1"
+  private_constant :BLESSED_BUNDLER_VERSIONS
+
   class GemfileParseError < BuildpackError
     def initialize(error)
-      msg = "There was an error parsing your Gemfile, we cannot continue\n"
+      msg = String.new("There was an error parsing your Gemfile, we cannot continue\n")
       msg << error
       super msg
     end
   end
 
-  VENDOR_URL         = LanguagePack::Base::VENDOR_URL                # coupling
-  DEFAULT_FETCHER    = LanguagePack::Fetcher.new(VENDOR_URL)         # coupling
-  BUNDLER_DIR_NAME   = LanguagePack::Ruby::BUNDLER_GEM_PATH          # coupling
-  BUNDLER_PATH       = File.expand_path("../../../../tmp/#{BUNDLER_DIR_NAME}", __FILE__)
-  GEMFILE_PATH       = Pathname.new "./Gemfile"
+  class UnsupportedBundlerVersion < BuildpackError
+    def initialize(version_hash, major)
+      msg = String.new("Your Gemfile.lock indicates you need bundler `#{major}.x`\n")
+      msg << "which is not currently supported. You can deploy with bundler version:\n"
+      version_hash.keys.each do |v|
+        msg << "  - `#{v}.x`\n"
+      end
+      msg << "\nTo use another version of bundler, update your `Gemfile.lock` to point\n"
+      msg << "to a supported version. For example:\n"
+      msg << "\n"
+      msg << "```\n"
+      msg << "BUNDLED WITH\n"
+      msg << "   #{version_hash["1"]}\n"
```

Comment:
why index 1 instead of 0?

Reply:
This is the major version. So this will be "1.15.2"